pyqtorch/analog.py

```diff
@@ -619,17 +628,14 @@ def forward(
         Returns:
             The transformed state.
         """
-        hamiltonian: torch.Tensor = self.create_hamiltonian(values)
-        time_evolution: torch.Tensor = (
-            values[self.time] if isinstance(self.time, str) else self.time
-        )  # If `self.time` is a string / hence, a Parameter,
-        # we expect the user to pass it in the `values` dict
+
+        evolved_op = self.tensor(values, None)
         return apply_operator(
             state=state,
-            operator=evolve(hamiltonian, time_evolution),
+            operator=evolved_op,
             qubits=self.qubit_support,
             n_qubits=len(state.size()) - 1,
-            batch_size=max(hamiltonian.shape[BATCH_DIM], len(time_evolution)),
```

Comment:
Did you check the logic of this `max` call here that you removed?

Now I don't remember exactly why it was needed, but it had something to do with evaluating a single fixed hamiltonian for a large number of different time values.

Since you removed it, and nothing breaks, either it was not really needed or it is untested 🤔

Reply:
@jpmoutinho  I could not really get the logic behind it but we are testing different batchdim and time_evo length in `test_hamiltonianevolution_with_types`, and different cases. All goes so I assumed it is necessary with the new way I implemented.

Reply:
Ok cool then let's assume it is fine 😅